docs/v2-system-addon/data_dictionary.md

```diff
@@ -152,7 +152,7 @@ Schema definitions/validations that can be used for tests can be found in `syste
 | `load_trigger_ts` | [Optional][Server Counter][Server Alert for too many omissions]  DOMHighResTimeStamp of the action perceived by the user to trigger the load of this page. | :one:
 | `load_trigger_type` | [Server Counter][Server Alert for too many omissions] Either ["menu_plus_or_keyboard", "unexpected"]. | :one:
 | `metadata_source` | [Optional] The source of which we computed metadata. Either (`MetadataService` or `Local` or `TippyTopProvider`). | :one:
-| `page` | [Required] Either ["NEW_TAB", "HOME"]. | :one:
+| `page` | [Optional] Either ["about:newtab", "about:home"]. | :one:
```

Comment:
Any reason for this change? Is it because it may fail to get the `url` in some cases?

Reply:
If we really can't get the `url` in certain cases (whether expected or not), I'd recommend to keep it as it is, even simply set it as "unknown" or "n/a" or any thing more specific, and optionally send an undesired event ping for later issue diagnose.

Our database doesn't support `ALTER COLUMN`, unfortunately, switching it from `not null` to `nullable` needs a lot of effort for the data migration. Usually we would have to populate it with the default zero value before database inserts as a workaround.

Reply:
@ncloudioj There are a couple of cases:

For some pings, it won't necessarily make sense to have one (e.g. PerfPings these days are likely to be mostly (entirely?) used to report on chrome performance where the cost is typically amortized across all the various pages and/or startup).  For that case, I think it'll be fairly straightforward to switch to "n/a" or something.

The bigger driver, though, is that the existing code sometimes creates pings without specifying what port they are attached to, meaning that we don't actually know, and up until now always just sent "about:newtab" because that was the only possibility..  This code is going to require some auditing/tweaking to handle these cases, but it shouldn't be too bad.  I'll give it a shot.

Out of curiosity, what database is it that doesn't support ALTER COLUMN?

Reply:
@dmose Oh, I see. BTW, the ETL jobs already treat `page` as an optional field for "perf", "undesired event", and "impression" pings.

> existing code sometimes creates pings without specifying what port they are attached to

Does this mean that we might not have `page` for the "session" and "event" pings *sometimes*?  We do check the existence of "page" for those two pings in the ETL. If so, we need to update the ETL rules for this change, i.e. use the default value "n/a" if it's missing.

> Out of curiosity, what database is it that doesn't support ALTER COLUMN?

We are using AWS Redshift, a column-oriented data warehouse. Unlike other DBMSs, it sacrifices some traditional features (e.g. ALTER COLUMN) to reach high capacity and query performance. Altering a column is usually done by "Rename the old col -> Create a new one -> Copy data over -> Drop the old one" :/